### PR TITLE
#159987652 highlight and comment on any text

### PR DESCRIPTION
**Pull request by gonuoha**
#### What does this PR do?
Adds ability to highlight and comment on any part of an article.

#### Description of Task to be completed?
- ability to highlight any text on article
- ability to comment and comment associated with highlighted text
- Design of interface for comment with highlighted text
- link highlighted comment to to target on click
#### How should this be manually tested?
- Clone the repo
- cd into the ‘thor-ah-frontend’
- run 'npm install'
- run 'npm start’
- navigate to an article page
- highlight any text and click comment icon when it appears
- type your comment in the comment box provided and click submit
- scroll to comments and click on comment with highlighted text to see where in the article it is associated.

#### What are the relevant pivotal tracker stories?
[#159987652](https://www.pivotaltracker.com/story/show/159987652)

### CODE REVIEW

File: src/actions/comments.js [hunk 1]
```diff
@@ -18,6 +18,32 @@ const createCommentError = (error) => ({
   payload: error,
 });
 
+
+// const createComment = (commentObject, articleSlug) => (dispatch) => {
+//   dispatch(createCommentRequest(true));
+//   const { token } = localStorage;
```
Comment (lauragift21) [inline src/actions/comments.js]: Is this block of code still required or should it be commented out like this?

File: src/actions/comments.js [hunk 2]
```diff
@@ -101,21 +102,20 @@ export const getArticleComments = (articleSlug) => (dispatch) => {
     });
 };
 
-const createComment = (commentData, articleSlug) => (dispatch) => {
+// comment object === {comment, articleBody,highlighted,cssId}   or {comment}
```
Comment (johnkingzy) [inline src/actions/comments.js]: I will suggest we take this line of code out

File: src/components/ArticleContent/ArticleContent.js [hunk 3]
```diff
@@ -29,8 +40,148 @@ class ArticleContent extends Component {
...
+    const element = document.getElementById(`${this.elementId}`);
+
+    if (element) {
+      const parent = element.parentNode;
+      while (element.firstChild) {
+        parent.insertBefore(element.firstChild, element);
+      }
+      parent.removeChild(element);
+
+      // set highlight stat
```
Comment (johnkingzy) [inline src/components/ArticleContent/ArticleContent.js]: Nitpick: state not stat

File: src/components/ArticleContent/ArticleContent.module.scss [hunk 4]
```diff
@@ -8,8 +8,66 @@
   font-weight: 500 !important;
   line-height: 2.2rem;
   font-style: normal !important;
+  // font-size: 18px;
```
Comment (johnkingzy) [inline src/components/ArticleContent/ArticleContent.module.scss]: we can take out this comment too

**Review by johnkingzy — APPROVED** [review]
LGTM

File: src/components/ArticleContent/ArticleContent.js [hunk 5]
```diff
@@ -29,8 +40,148 @@ class ArticleContent extends Component {
...
+    const { promptStyle } = this.state
+    this.setState({
+      promptStyle: {
+        ...promptStyle,
+        display: 'none'
+      }
+    })
+  }
+
+  // Hilighted comment Component
```
Comment (d-beloved) [inline src/components/ArticleContent/ArticleContent.js]: Nitpick: Highlighted not Hilighted

**Review by d-beloved — APPROVED** [review]
LGTM

File: src/components/ArticleContent/ArticleContent.module.scss [hunk 6]
```diff
@@ -8,8 +8,66 @@
...
+  // z-index: 5;
```
Comment (lauragift21) [inline src/components/ArticleContent/ArticleContent.module.scss]: nitpick: we can take out this comment

Comment (lauragift21) [inline src/components/ArticleContent/ArticleContent.module.scss]: nitpick: we can take out this comment

**Review by lauragift21 — APPROVED** [review]
LGTM

**Pull request merged**